File: src/backend/schemas/breach.py
"""Pydantic schemas for breach_notifications (M10d)."""

from __future__ import annotations

from datetime import datetime
from enum import Enum
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class BreachStatus(str, Enum):
    DETECTED = "DETECTED"
    DPO_NOTIFIED = "DPO_NOTIFIED"
    NPC_SUBMITTED = "NPC_SUBMITTED"
    CLOSED = "CLOSED"


class BreachResponse(BaseModel):
    model_config = {"from_attributes": True}

    breach_id: int
    threat_log_id: int
    detected_at: datetime
    npc_deadline_at: datetime
    status: BreachStatus
    affected_systems: Optional[str] = None
    data_scope: Optional[str] = None
    notes: Optional[str] = None
    reported_by: Optional[UUID] = None
    npc_submitted_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime


class BreachUpdate(BaseModel):
    status: Optional[BreachStatus] = None
    affected_systems: Optional[str] = None
    data_scope: Optional[str] = None
    notes: Optional[str] = None

File: src/frontend/src/lib/api/breach.ts
import { apiFetch } from './transport';

export type BreachStatus = 'DETECTED' | 'DPO_NOTIFIED' | 'NPC_SUBMITTED' | 'CLOSED';

export interface Breach {
    breach_id: number;
    threat_log_id: number;
    detected_at: string;
    npc_deadline_at: string;
    status: BreachStatus;
    affected_systems: string | null;
    data_scope: string | null;
    notes: string | null;
    reported_by: string | null;
    npc_submitted_at: string | null;
    created_at: string;
    updated_at: string;
}

export interface BreachUpdate {
    status?: BreachStatus;
    affected_systems?: string;
    data_scope?: string;
    notes?: string;
}

export async function fetchBreaches(): Promise<Breach[]> {
    return apiFetch<Breach[]>('/api/admin/breach');
}

export async function updateBreach(id: number, payload: BreachUpdate): Promise<Breach> {
    return apiFetch<Breach>(`/api/admin/breach/${id}`, {
        method: 'PATCH',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify(payload),
    });
}
